### PR TITLE
Refactor: Simplify code in serialization

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/serialization/serde.py [hunk 1]
```diff
@@ -300,14 +300,13 @@ def _stringify(classname: str, version: int, value: T | None) -> str:
 
     s = f"{classname}@version={version}("
     if isinstance(value, _primitives):
-        s += f"{value})"
```
Comment (potiuk) [inline airflow/serialization/serde.py:303]: Hmm. That sounds like a bug .

Comment (potiuk) [reply airflow/serialization/serde.py:303]: Ah. I see.

**Review by potiuk — APPROVED** [review]

File: airflow/serialization/helpers.py [hunk 2]
```diff
@@ -32,9 +32,10 @@ def serialize_template_field(template_field: Any) -> str | dict | list | int | f
     def is_jsonable(x):
         try:
             json.dumps(x)
-            return True
         except (TypeError, OverflowError):
             return False
+        else:
```
Comment (uranusjr) [inline airflow/serialization/helpers.py:37]: Even the else is arguably not needed

File: airflow/serialization/serde.py [hunk 3]
```diff
@@ -286,7 +286,7 @@ def _convert(old: dict) -> dict:
 
 
 def _match(classname: str) -> bool:
-    return any(p.match(classname) is not None for p in _get_patterns())
+    return any(p.match(classname) for p in _get_patterns())
```
Comment (uranusjr) [inline airflow/serialization/serde.py]: I prefer the `is not None` variant for explicitness. There’s not perceivable performance difference also.

File: airflow/serialization/helpers.py (hunk 2 above)
Comment (eumiro) [reply airflow/serialization/helpers.py:37]: It isn't, but it aligns nicely with the `return False` partner.

File: airflow/serialization/serde.py (hunk 3 above)
Comment (eumiro) [reply airflow/serialization/serde.py]: Okay, will revert.

File: airflow/serialization/helpers.py (hunk 2 above)
Comment (potiuk) [reply airflow/serialization/helpers.py:37]: Agree. We have not used too much of the `try/except/else` but I started to like it, it's much more explicit than return false in this case.

**Review by potiuk — APPROVED** [review]

**Pull request merged**